crates/matrix-sdk-crypto-ffi/src/machine.rs

```diff
@@ -1336,16 +1336,29 @@ impl OlmMachine {
     /// key.
     pub fn save_recovery_key(
         &self,
-        key: Option<String>,
+        key: Option<Arc<BackupRecoveryKey>>,
         version: Option<String>,
     ) -> Result<(), CryptoStoreError> {
-        let key = key.map(|k| RecoveryKey::from_base64(&k)).transpose().ok().flatten();
+        let key = key.map(|k| {
+            // We need to clone here due to FFI limitations but RecoveryKey does
```

Comment:
But couldn't we implement
```
pub fn from(other: &Arc<RecoveryKey>) -> RecoveryKey {
   RecoveryKey { inner: other.inner.clone() }
}
```
on `RecoveryKey` and use that here? Wouldn't that be more efficient?

It would prevent regular `clone` on `RecoveryKey` but allow you to create a new one from an old one without jumping through encode -> decode hoops and ensure everything stays `zeroize`. What am I missing?

Reply:
I think you may be misreading somewhere; it is `RecoveryKey` which doesn't support clone also, not just `BackupRecoveryKey`. The intention is probably to stave off people mistakenly cloning something we shouldn't clone.